src/library_webgl.js

```diff
@@ -3334,7 +3423,48 @@ var LibraryGL = {
     if (length) {{{ makeSetValue('length', '0', 'numBytesWrittenExclNull', 'i32') }}};
   },
 
+#if GL_EXPLICIT_UNIFORM_BINDING
+  // Applies the explicit sampler and ubo binding locations to the current program. This is done
+  // lazily the first time we glUseProgram() so that parallel shader compilation is not disturbed.
+  $webglApplyExplicitProgramBindings: function() {
+    var p = GLctx.currentProgram;
+    if (!p.explicitProgramBindingsApplied) {
+#if MAX_WEBGL_VERSION >= 2
+#if MIN_WEBGL_VERSION < 2
+      if (GL.currentContext.version >= 2) {
+#endif
+        Object.keys(p.explicitUniformBindings).forEach((ubo) => {
+          var bindings = p.explicitUniformBindings[ubo];
```

Comment:
nit: perhaps simply:
```suggestion
        Object.entries(p.explicitUniformBindings).forEach(([ubo, bindings]) => {
```
or even
```suggestion
        Object.entries(p.explicitUniformBindings).forEach(([ubo, [bindingsStart, bindingsLength] ]) => {
```
although I don't know if the array-argument-destructuring can be used in Emscripten.

Reply:
That would be nice, though we have been strictly ES5 so far. I doubt that anyone using this would target ES5 browsers, but there are outdated JS tools that people use (terser/uglify/acorn/closure), so I want to introduce ES6 explicitly as its own deliberate PR in the future.